src/common/containers/Profile.js

```diff
@@ -1,28 +1,102 @@
+import { object, func } from 'prop-types';
+import { withRouter } from 'react-router-dom';
 import Helmet from 'react-helmet';
 import { connect } from 'react-redux';
-import { withRouter } from 'react-router';
+import { withStyles } from '@material-ui/core/styles';
 import Paper from '@material-ui/core/Paper';
 
+import RecentCardsCarousel from '../components/cards/RecentCardsCarousel';
+import Title from '../components/Title';
+import * as collectionActions from '../actions/collection';
+
 export function mapStateToProps(state) {
   return {
-    loggedIn: state.global.user !== null
+    user: state.global.user
   };
 }
 
-const About = function (props) {
-  return (
-    <div style={{margin: '48px 72px'}}>
-      <Helmet title="Profile"/>
+export function mapDispatchToProps(dispatch) {
+  return {
+    onOpenForEditing: (card) => {
+      dispatch(collectionActions.openForEditing(card));
+    }
+  };
+}
 
-      <div style={{display: 'flex', justifyContent: 'stretch'}}>
-        <Paper style={{padding: '5px 20px'}}>
-          <h2>Profile</h2>
-          <p>TODO: Stuff goes here.</p>
-        </Paper>
-      </div>
-    </div>
-  );
+const styles = {
+  container: {
+    margin: 20
+  },
+  paperContainer: {
+    display: 'flex',
+    justifyContent: 'space-between'
+  },
+  gridItem: {
+    width: 'calc((100% / 3) - (40px / 3))',
+    height: 600
+  },
+  recentCardsContainer: {
+    marginTop: 20,
+    padding: '5px 25px'
+  }
 };
 
-export default withRouter(connect(mapStateToProps)(About));
+class Profile extends Component {
+  constructor(props) {
+    super(props);
+
+    if (!this.userFromPath) {
+      props.history.push('/');
+    }
+
+    this.state = {
+      userId: this.userFromPath
+    };
+  }
+
+  static propTypes = {
+    user: object,
+    classes: object,
+    history: object,
+
+    onOpenForEditing: func
+  }
+
+  get userFromPath() {
+    const { history } = this.props;
+    return history.location.pathname.split('/')[2];
+  }
+
+  render() {
+    const { user, classes } = this.props;
+    const { userId } = this.state;
+
+    const title = user && user.displayName ? `${user.displayName}'s Profile` : 'Profile';
+
+    return (
+      <div>
+        <Helmet title="Profile"/>
+        <Title text={title} />
+
+        <div className={classes.container}>
+          <div className={classes.paperContainer}>
+            <Paper className={classes.gridItem} />
+            <Paper className={classes.gridItem} />
+            <Paper className={classes.gridItem} />
+          </div>
+          <Paper className={classes.recentCardsContainer}>
+            { userId &&
+              <RecentCardsCarousel
+                userId={userId}
+                history={this.props.history}
+                onOpenForEditing={this.props.onOpenForEditing} />
+            }
+          </Paper>
+        </div>
+      </div>
+    );
+  }
+}
+
+export default withRouter(connect(mapStateToProps, mapDispatchToProps)(withStyles(styles)(Profile)));
```

Comment:
Maybe put a TODO to replace this grossness with decorators once we get that Babel extension working.

Reply:
Ugh yeah that's fair